src/elm/Page/Register.elm

```diff
@@ -683,64 +793,244 @@ update maybeInvitation msg model guest =
-                , problems = []
+                | status =
+                    case model.status of
+                        LoadedInvite invitation ->
+                            LoadedAll invitation country
+
+                        FailedInvite err ->
+                            FailedInvite err
+
+                        NotFound ->
+                            NotFound
+
+                        _ ->
+                            LoadedCountry country
             }
+                |> UR.init
+
+        CompletedLoadCountry (Ok Nothing) ->
+            UR.init { model | status = NotFound }
+
+        CompletedLoadCountry (Err error) ->
+            { model | status = FailedCountry error }
+                |> UR.init
+                |> UR.logGraphqlError msg error
+
+
+type alias SignUpResponse =
+    { reason : String
+    , status : SignUpStatus.SignUpStatus
+    }
+
+
+formTypeToAccountCmd : Shared -> String -> FormType -> Cmd Msg
+formTypeToAccountCmd shared key formType =
+    let
+        cmd obj =
+            Api.Graphql.mutation shared
+                (Mutation.signUp
+                    { input =
+                        InputObject.buildSignUpInput
+                            obj
+                            (\x -> x)
+                    }
+                    (Graphql.SelectionSet.succeed SignUpResponse
+                        |> with Cambiatus.Object.SignUp.reason
+                        |> with Cambiatus.Object.SignUp.status
+                    )
+                )
+                CompletedSignUp
     in
-    case msg of
-        EnteredUsername str ->
-            updateModel { form | username = str }
+    case formType of
+        Juridical form ->
+            cmd { account = form.account, email = form.email, name = form.name, publicKey = key }
+
+        Natural form ->
+            cmd { account = form.account, email = form.email, name = form.name, publicKey = key }
+
+        Default form ->
+            cmd { account = form.account, email = form.email, name = form.name, publicKey = key }
 
-        EnteredEmail str ->
-            updateModel { form | email = str }
+        None ->
+            Cmd.none
 
-        EnteredAccount str ->
-            updateModel { form | account = str |> String.toLower }
+
+redirectCmd : Shared -> Cmd Msg
+redirectCmd shared =
+    Route.replaceUrl shared.navKey (Route.Login Nothing)
+
+
+formTypeToKycCmd : Shared -> FormType -> Cmd Msg
+formTypeToKycCmd shared formType =
+    let
+        cmd : InputObject.KycDataUpdateInputRequiredFields -> Cmd Msg
+        cmd obj =
+            Api.Graphql.mutation shared
+                (Mutation.upsertKyc
+                    { input = InputObject.buildKycDataUpdateInput obj }
+                    Graphql.SelectionSet.empty
+                )
+                CompletedKycUpsert
+    in
+    case formType of
+        Juridical form ->
+            cmd
+                { accountId = form.account
+                , countryId = "1"
```

Comment:
@RodrigoLeiteF  there will be a type mismatch after merging the latest changes from the `kyc` branch since the type for `county_id` and other address fields were changed to `:id` instead of `:string`.

I already had this mismatch somewhere, it can be fixed just by importing `import Cambiatus.Scalar exposing (Id(..))` and using `countryId = Id "1"`.